controllers/operatorpolicy_controller.go

```diff
@@ -193,79 +200,142 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+// why the desired subscription can't be determined.
+func buildSubscription(
+	policy *policyv1beta1.OperatorPolicy, defaultNS string,
+) (*operatorv1alpha1.Subscription, error) {
+	subscription := new(operatorv1alpha1.Subscription)
+
+	sub := make(map[string]interface{})
+
+	err := json.Unmarshal(policy.Spec.Subscription.Raw, &sub)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal from spec.subscription: %w", err)
+	}
+
+	ns, ok := sub["namespace"].(string)
+	if !ok {
+		if defaultNS == "" {
+			return nil, fmt.Errorf("namespace is required in spec.subscription")
 		}
 
-		catalogName = sub.Spec.CatalogSource
-		catalogNS = sub.Spec.CatalogSourceNamespace
-	} else {
-		catalogName = subscription.Spec.CatalogSource
-		catalogNS = subscription.Spec.CatalogSourceNamespace
+		ns = defaultNS
 	}
 
-	// Check if CatalogSource exists
-	foundCatalogSrc, err := r.DynamicWatcher.Get(watcher, catalogSrcGVK,
-		catalogNS, catalogName)
+	if validationErrs := validation.IsDNS1123Label(ns); len(validationErrs) != 0 {
+		return nil, fmt.Errorf("the namespace '%v' used for the subscription is not a valid namespace identifier", ns)
+	}
+
+	spec := new(operatorv1alpha1.SubscriptionSpec)
+
+	err = json.Unmarshal(policy.Spec.Subscription.Raw, spec)
 	if err != nil {
-		return fmt.Errorf("error getting CatalogSource: %w", err)
+		return nil, fmt.Errorf("unable to unmarshal from spec.subscription: %w", err)
 	}
 
-	isMissing := foundCatalogSrc == nil
-	isUnhealthy := isMissing
+	subscription.SetGroupVersionKind(subscriptionGVK)
+	subscription.ObjectMeta.Name = spec.Package
+	subscription.ObjectMeta.Namespace = ns
+	subscription.Spec = spec
 
-	if !isMissing {
-		// CatalogSource is found, initiate health check
-		catalogSrcUnstruct := foundCatalogSrc.DeepCopy()
-		catalogSrc := new(operatorv1alpha1.CatalogSource)
+	// If the policy is in `enforce` mode and the allowed CSVs are restricted,
+	// the InstallPlanApproval will be set to Manual so that upgrades can be controlled.
+	if policy.Spec.RemediationAction.IsEnforce() && len(policy.Spec.Versions) > 0 {
+		subscription.Spec.InstallPlanApproval = operatorv1alpha1.ApprovalManual
+	}
 
-		err := runtime.DefaultUnstructuredConverter.
-			FromUnstructured(catalogSrcUnstruct.Object, catalogSrc)
-		if err != nil {
-			return fmt.Errorf("error converting the retrieved CatalogSource to the Go type: %w", err)
-		}
+	return subscription, nil
+}
 
-		if catalogSrc.Status.GRPCConnectionState == nil {
-			// Unknown State
-			err := r.updateStatus(ctx, policy, catalogSourceUnknownCond, catalogSrcUnknownObj(catalogName, catalogNS))
-			if err != nil {
-				return fmt.Errorf("error retrieving the status for a CatalogSource: %w", err)
-			}
+// buildOperatorGroup bootstraps the OperatorGroup spec defined in the operator policy
+// with the apiversion and kind in preparation for resource creation
+func buildOperatorGroup(
+	policy *policyv1beta1.OperatorPolicy, defaultNS string,
```

Comment:
```suggestion
	policy *policyv1beta1.OperatorPolicy, namespace string,
```

Reply:
It would still use the namespace specified by the user if one is provided, so I feel like `defaultNS` is a more accurate name. But, thinking about it, we should probably stop and alert the user if the specified namespace for the operator group does not match what the subscription will use.

Reply:
In which case, the parameter should just be called `namespace` 😆